helm/cluster/templates/clusterapi/workers/machinepool.yaml

```diff
@@ -1,11 +1,11 @@
 {{- if $.Values.providerIntegration.resourcesApi.machinePoolResourcesEnabled }}
 {{- if eq $.Values.providerIntegration.resourcesApi.nodePoolKind "MachinePool" }}
-{{- range $nodePoolName, $nodePoolConfig := $.Values.global.nodePools }}
+{{- range $nodePoolName, $nodePoolConfig := $.Values.global.nodePools | default $.Values.providerIntegration.workers.defaultNodePools }}
```

Comment:
This is a bit confusing to me, but why do we sometimes use nodePool and sometimes machinePool as variables.

Could we try to align here if that is possible?

Reply:
Yeah, good point, will check and align.

Reply:
I renamed `$machinePools` to `$nodePools` in one of the helper files.

So now we have:
- `$nodePools` as a variable for our Helm values, as we are using the term node pools in our API.
- We have MachinePools as a specific node pools implementation, so the term MachinePool is used in places where it refers to CAPI's MachinePool CR.
- We will at some point (soon) add MachineDeployments as another node pools implementation, and the naming should work like above - `$nodePools` as a variable name and MachineDeployment when referring to CAPI MachineDeployments specifically.

Reply:
Wonderful ❤️ ❤️ ❤️